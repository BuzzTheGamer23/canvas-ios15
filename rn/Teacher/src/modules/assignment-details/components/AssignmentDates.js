/**
 * @flow
 *
 * - When single due date assigned to everyone, display due date, assignees (i.e., "Everyone"), and available from/to dates
 * - When single due date assigned to one person, display due date, assignee (e.g., "Tom Smith"), and available from/to dates
 * - When single due date assigned to one section or group, display due date, assignee (e.g., "Student Group 2") and available from/to dates
 * - When single due date assigned to more than one person and less than everyone, display due date, assignees (e.g., "20 people"), and available from/to dates
 * - When multiple due dates, display "Multiple due dates"
 * - Assigned to multiple sections or groups, treat as multiple due dates
 * - Assigned to at least one section or group and at least one person, treat as multiple due dates
 * - if no Available To date, display double dash
 * - if available to date is in the past, display "Availability: Closed" rather than available from/to dates
 * - If multiple due dates, don't show closed unless all availability dates are past
 */

import React, { Component } from 'react'

import {
  View,
  StyleSheet,
} from 'react-native'
import i18n from 'format-message'
import AssignmentDates from '../../../common/AssignmentDates'
import { formattedDueDate } from '../../../common/formatters'
import { Text } from '../../../common/text'

type Props = {
  assignment: Assignment,
}

export default class DueDates extends Component<any, Props, any> {

  renderMultipleDueDates (): React.Element<View> {
    const title = i18n({
      default: 'Multiple Due Dates',
      description: 'Assignment details information about due dates, when there are more than one due date to display',
    })

    return <View style={styles.container}>
             <Text>{title}</Text>
           </View>
  }

  renderAvailability (dates: AssignmentDates): React.Element<View> {
    if (dates.availabilityClosed()) {
      const availabilityClosedTitle = i18n({
        default: 'Availability:',
        description: 'Assignment due date "Available to:" field',
      })

      const availabilityClosedInfo = i18n({
        default: 'Closed',
        description: 'Assignment due date "Available to:" field',
      })

      return <View>
               <Text style={styles.textContainer}>
                 <Text style={styles.descriptionText}>{availabilityClosedTitle}</Text>
                 <Text style={styles.infoText}>{` ${availabilityClosedInfo}`}</Text>
               </Text>
             </View>
    }

    const availableFromTitle = i18n({
      default: 'Available from:',
      description: 'Assignment due date "Available from:" field',
    })

    const availableToTitle = i18n({
      default: 'Available to:',
      description: 'Assignment due date "Available to:" field',
    })

    const availableFrom = dates.bestAvailableFrom()
    const availableTo = dates.bestAvailableTo()
    const availableFromText = availableFrom ? formattedDueDate(availableFrom) : '--'
    const availableToText = availableTo ? formattedDueDate(availableTo) : '--'

    let availableFromAccessibilityLabel
    let availableToAccessibilityLabel

    if (!availableFrom) {
      availableFromAccessibilityLabel = i18n({
        default: 'No available from date set.',
        description: 'Accessibility label for when there is no available from date set',
      })
    }

    if (!availableTo) {
      availableToAccessibilityLabel = i18n({
        default: 'No available to date set.',
        description: 'Accessibility label for when there is no available to date set',
      })
    }

    return (<View>
              <View accessible={true} accessibilityLabel={availableFromAccessibilityLabel}>
                <Text style={styles.textContainer}>
                  <Text style={styles.descriptionText}>{availableFromTitle}</Text>
                  <Text style={styles.infoText}>{` ${availableFromText}`}</Text>
                </Text>
              </View>
              <View accessible={true} accessibilityLabel={availableToAccessibilityLabel}>
                <Text style={styles.textContainer}>
                  <Text style={styles.descriptionText}>{availableToTitle}</Text>
                  <Text style={styles.infoText}>{` ${availableToText}`}</Text>
                </Text>
              </View>
          </View>)
  }

  renderSingleDueDate (dates: AssignmentDates): ReactElement<View> {
    const dueTitle = i18n({
      default: 'Due:',
      description: 'Assignment due date "Due:" field. Looks something like this when finished: "Due: March 13, 2019"',
    })

    const forTitle = i18n({
      default: 'For:',
      description: 'Assignment due date "For:" field',
    })

    const dueAtValue = dates.bestDueAt() ? formattedDueDate(dates.bestDueAt()) : '--'
    const availability = this.renderAvailability(dates)
    let dueAtAccessibilityLabel

    if (!dates.bestDueAt()) {
      dueAtAccessibilityLabel = i18n({
        default: 'No due date set.',
        description: 'Accessibility label for when no due date is set for an assignment',
      })
    }

    return (<View style={styles.container}>
              <View accessible={true} accessibilityLabel={dueAtAccessibilityLabel}>
                <Text style={styles.textContainer}>
                  <Text style={styles.descriptionText}>{dueTitle}</Text>
                  <Text style={styles.infoText}>{` ${dueAtValue}`}</Text>
                </Text>
              </View>
              <Text style={styles.textContainer}>
                <Text style={styles.descriptionText}>{forTitle}</Text>
                <Text style={styles.infoText}>{` ${dates.bestDueDateTitle()}`}</Text>
              </Text>
              {availability}
           </View>)
  }

  render (): React.Element<View> {
    const dates = new AssignmentDates(this.props.assignment)

    if (dates.hasMultipleDueDates()) {
      return this.renderMultipleDueDates()
    }

    return this.renderSingleDueDate(dates)
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  textContainer: {
    paddingTop: 2,
  },
  descriptionText: {
    fontWeight: '600',
  },
  infoText: {
  },
})
