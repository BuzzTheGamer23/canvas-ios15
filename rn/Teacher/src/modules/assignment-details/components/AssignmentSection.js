/**
 * @flow
 */

import React, { Component } from 'react'
import { Text } from '../../../common/text'
import color from '../../../common/colors'
import DisclosureIndicator from '../../../common/components/DisclosureIndicator'

import {
  View,
  TouchableHighlight,
  Image,
  StyleSheet,
} from 'react-native'

export default class AssignmentSection extends Component {

  render (): ReactElement<*> {
    let dividerStyle = {}
    let headerStyle = {}
    if (!this.props.isFirstRow) {
      dividerStyle = assignmentSectionStyles.divider
      headerStyle = assignmentSectionStyles.header
    }

    return (
      <TouchableHighlight onPress={this.props.onPress}>
        <View style={[assignmentSectionStyles.container, this.props.style]} >
          <View style={dividerStyle}></View>
          <View style={assignmentSectionStyles.innerContainer}>
            <View style={{ flex: 1 }}>
              <View style={assignmentSectionStyles.titleContainer}>
                {
                  this.props.image && <Image source={this.props.image} style={assignmentSectionStyles.image} />
                }
                <Text style={headerStyle}>{this.props.title}</Text>
              </View>
              {this.props.children}
            </View>

            { this.props.showDisclosureIndicator && <View style={assignmentSectionStyles.disclosureIndicatorContainer} >
                                                      <DisclosureIndicator style={{ right: 0 }} />
                                                    </View> }
          </View>
        </View>
      </TouchableHighlight>
    )
  }
}

const assignmentSectionStyles = StyleSheet.create({
  container: {
    flex: 1,
    paddingLeft: global.style.defaultPadding,
    paddingRight: global.style.defaultPadding,
    paddingBottom: global.style.defaultPadding,
    backgroundColor: 'white',
  },
  innerContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  divider: {
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: color.grey2,
    paddingBottom: global.style.defaultPadding,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 4,
  },
  header: {
    color: color.grey4,
    fontWeight: '500',
    fontSize: 16,
    paddingTop: 2,
  },
  image: {
    tintColor: color.grey4,
    height: 19,
    width: 19,
    marginRight: 5,
  },
  disclosureIndicatorContainer: {
    flex: 0,
    width: 10,
    paddingRight: 2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
})
