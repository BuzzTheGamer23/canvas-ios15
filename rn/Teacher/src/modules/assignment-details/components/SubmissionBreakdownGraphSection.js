/**
 * @flow
 */

import React, { Component } from 'react'
import SubmissionGraph from '../../submissions/SubmissionGraph'
import { connect } from 'react-redux'
import SubmissionActions from '../../submissions/list/actions'

import {
  View,
  StyleSheet,
  ActivityIndicator,
  LayoutAnimation,
  TouchableOpacity,
  Text,
} from 'react-native'
import i18n from 'format-message'

export type SubmissionBreakdownGraphSectionProps = {
  courseID: string,
  assignmentID: string,
  style: any,
  onPress: (string) => void,
  graded: number,
  ungraded: number,
  not_submitted: number,
  submissionTotalCount: number,
  refreshSubmissionSummary: Function,
  submissionTypes: string[],
}

export type SubmissionBreakdownGraphSectionInitProps = {
  courseID: string,
  assignmentID: string,
}

export class SubmissionBreakdownGraphSection extends Component<any, SubmissionBreakdownGraphSectionProps, any> {
  componentDidMount () {
    refreshSubmissionList(this.props)
  }

  componentWillUpdate () {
    LayoutAnimation.easeInEaseOut()
  }

  renderNoSubmissions () {
    let noSubmissionsMessage = i18n('Tap to view submissions list.')
    return (
        <View style={[style.container, this.props.style]} accessible={false}>
          <View style={style.paperOnlyContainer} accessible={false} accessibilityLabel={noSubmissionsMessage}>
            <View style={style.paperOnlyContainer}>
              <TouchableOpacity underlayColor='#eeeeee00' style={style.common}
                                key={`submission_dial_highlight_no_submissions`}
                                testID={`assignment-details.submission-breakdown-graph-section.no-submissions`}
                                onPress={() => this.props.onPress('') } accessible={false}>
                <Text style={style.paperOnlyContainerLabel} accessible={false}>{noSubmissionsMessage}</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
    )
  }

  render () {
    let gradedLabel = i18n('Graded')

    let ungradedLabel = i18n('Ungraded')

    let notSubmittedLabel = i18n('Not Submitted')

    let labels = [gradedLabel, ungradedLabel, notSubmittedLabel]

    let ids = ['graded', 'ungraded', 'not-submitted']

    if (this.props.pending) {
      return <View style={style.loadingWrapper}><ActivityIndicator /></View>
    }

    let totalStudents = this.props.submissionTotalCount
    let graded = this.props.graded
    let ungraded = this.props.ungraded
    let notSubmitted = this.props.not_submitted
    let submissionTypes = this.props.submissionTypes || []

    let noSubmissions = submissionTypes.includes('none')
    if (noSubmissions) { return this.renderNoSubmissions() }

    let paperOnly = submissionTypes.includes('on_paper')
    let paperOnlyMessage = i18n({
      default: `{
        count, plural,
        one {There is # assignee without a grade.}
        other {There are # assignees without grades.}
      }`,
      message: 'Number of assignees without grades.',
    }, { count: ungraded + notSubmitted })

    let data = [graded]
    if (!paperOnly) {
      data.push(ungraded)
      data.push(notSubmitted)
    }

    return (
      <View style={[style.container, this.props.style]}>
        {data.map((item, index) =>
          <TouchableOpacity underlayColor='#eeeeee00' style={style.common} key={`submission_dial_highlight_${index}`}
                              testID={`assignment-details.submission-breakdown-graph-section.${ids[index]}-dial`} onPress={() => this.onPress(index) } accessibilityTraits='button'>
            <View>
              <SubmissionGraph
                label={labels[index]}
                total={totalStudents}
                current={data[index]}
                key={index}
                testID={`${ids[index]}`}
              />
            </View>
          </TouchableOpacity>
        )}

        {paperOnly &&
          <View style={style.paperOnlyContainer} accessible={true} accessibilityLabel={paperOnlyMessage}>
            <View style={style.paperOnlyContainer} >
            <Text style={style.paperOnlyContainerLabel}>{paperOnlyMessage}</Text>
            </View>
          </View>
        }

      </View>
    )
  }

  onPress (itemIndex: number) {
    let graded = 0
    let ungraded = 1
    let notSubmitted = 2

    if (!this.props.onPress) return

    switch (itemIndex) {
      case graded:
        this.props.onPress('graded')
        break
      case ungraded:
        this.props.onPress('notgraded')
        break
      case notSubmitted:
        this.props.onPress('notsubmitted')
        break
    }
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  loadingWrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  common: {
    flex: 1,
  },
  paperOnlyContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export function refreshSubmissionList (props: SubmissionBreakdownGraphSectionProps): void {
  props.refreshSubmissionSummary(props.courseID, props.assignmentID)
}

export function mapStateToProps (state: AppState, ownProps: SubmissionBreakdownGraphSectionInitProps): any {
  const assignment = state.entities.assignments[ownProps.assignmentID]
  let pending = 0
  let submissionTotalCount = 0
  let summary = { graded: 0, ungraded: 0, not_submitted: 0 }
  if (assignment && assignment.submissionSummary) {
    summary = assignment.submissionSummary.data
    submissionTotalCount = summary.graded + summary.ungraded + summary.not_submitted
    pending = assignment.pending || assignment.submissionSummary.pending
  } else {
    pending = true
  }

  return {
    ...summary,
    submissionTotalCount,
    pending: pending,
  }
}

const Connected = connect(mapStateToProps, { ...SubmissionActions })(SubmissionBreakdownGraphSection)
export default (Connected: any)
