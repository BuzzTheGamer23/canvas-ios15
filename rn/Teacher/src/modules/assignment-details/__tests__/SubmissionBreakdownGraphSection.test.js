/**
 * @flow
 */

import { LayoutAnimation } from 'react-native'
import React from 'react'
import { SubmissionBreakdownGraphSection, mapStateToProps } from '../components/SubmissionBreakdownGraphSection'
import renderer from 'react-test-renderer'
import explore from '../../../../test/helpers/explore'
const template = {
  ...require('../../../api/canvas-api/__templates__/assignments'),
  ...require('../../../api/canvas-api/__templates__/users'),
  ...require('../../../api/canvas-api/__templates__/course'),
  ...require('../../../api/canvas-api/__templates__/submissions'),
  ...require('../../../redux/__templates__/app-state'),
}
jest.mock('LayoutAnimation', () => ({
  create: jest.fn(),
  configureNext: jest.fn(),
  easeInEaseOut: jest.fn(),
  Types: { linear: null },
  Properties: { opacity: null },
  onPress: jest.fn(),
}))
jest.mock('TouchableOpacity', () => 'TouchableOpacity')

let course: any = template.course()
let assignment: any = template.assignment()

let defaultProps = {}

beforeEach(() => {
  const a = template.submission({ id: 1, grade: '95' })
  const b = template.submission({ id: 2, grade: 'ungraded' })
  const c = template.submission({ id: 3, grade: 'not_submitted' })
  const submissions = [a, b, c]

  defaultProps = {
    courseID: course.id,
    assignmentID: assignment.assignmentID,
    refreshSubmissions: jest.fn(),
    refreshGradeableStudents: jest.fn(),
    refreshAssignment: jest.fn(),
    submissions,
    submissionTotalCount: submissions.length,
    pending: 0,
    refresh: jest.fn(),
    refreshing: false,
    onPress: jest.fn(),
    refreshSubmissionSummary: jest.fn(),
  }
})

test('render', () => {
  let tree = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render 0 submissions', () => {
  defaultProps.submissions = []
  defaultProps.submissionTotalCount = 0
  let tree = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render multiple data points ', () => {
  const a = template.submission({ id: 1, grade: '95' })
  const b = template.submission({ id: 2, grade: 'A' })
  const c = template.submission({ id: 3, grade: 'B-' })
  const d = template.submission({ id: 4, grade: 'ungraded' })
  const e = template.submission({ id: 5, grade: 'ungraded' })
  const f = template.submission({ id: 6, grade: 'not_submitted' })
  const submissions = [a, b, c, d, e, f]
  defaultProps.submissions = submissions
  defaultProps.submissionTotalCount = submissions.length

  const tree = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render loading with null submissions', () => {
  defaultProps.submissions = null
  let tree = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render loading with pending set', () => {
  defaultProps.pending = 1
  let tree = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress is called graded dial', () => {
  testDialOnPress('assignment-details.submission-breakdown-graph-section.graded-dial', 'graded')
})

test('onPress is called ungraded dial', () => {
  testDialOnPress('assignment-details.submission-breakdown-graph-section.ungraded-dial', 'notgraded')
})

test('onPress is called not_submitted dial', () => {
  testDialOnPress('assignment-details.submission-breakdown-graph-section.not-submitted-dial', 'notsubmitted')
})

function testDialOnPress (expectedID: string, expectedValueParameter: string) {
  let component = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  )
  let dial: any = explore(component.toJSON()).selectByID(expectedID)
  dial.props.onPress()
  expect(defaultProps.onPress).toBeCalledWith(expectedValueParameter)
}

test('mapStateToProps with an assignment id', () => {
  const course = template.course()
  const assignment = template.assignment()

  const u1 = template.user({
    id: '1',
  })
  const s1 = template.submission({
    id: '1',
    assignment_id: assignment.id,
    user_id: u1.id,
    workflow_state: 'unsubmitted',
  })

  const appState = template.appState({
    entities: {
      assignments: {
        [assignment.id]: {
          assignment,
          submissions: { refs: [s1.id], pending: 0 },
          submissionSummary: { data: { graded: 0, ungraded: 0, not_submitted: 1 }, error: null, pending: 0 },
          gradeableStudents: { refs: ['1'], pending: 0 },
        },
      },
    },
  })

  const result = mapStateToProps(appState, { courseID: course.id, assignmentID: assignment.id })
  expect(result).toMatchObject({
    submissionTotalCount: 1,
    graded: 0,
    ungraded: 0,
    not_submitted: 1,
    pending: 0,
  })
})

test('mapStateToProps no datas', () => {
  const course = template.course()
  const assignment = template.assignment()

  const appState = template.appState()

  const result = mapStateToProps(appState, { courseID: course.id, assignmentID: assignment.id })
  expect(result).toMatchObject({
    submissionTotalCount: 0,
    graded: 0,
    ungraded: 0,
    not_submitted: 0,
    pending: true,
  })
})

test('misc functions', () => {
  defaultProps.onPress = null
  let instance = renderer.create(
    <SubmissionBreakdownGraphSection {...defaultProps} />
  ).getInstance()

  // Make sure this doesn't explode when there isn't an onPress handler
  instance.onPress(null)

  // Make sure animation is called
  instance.componentWillUpdate()
  expect(LayoutAnimation.easeInEaseOut).toHaveBeenCalled()
})
