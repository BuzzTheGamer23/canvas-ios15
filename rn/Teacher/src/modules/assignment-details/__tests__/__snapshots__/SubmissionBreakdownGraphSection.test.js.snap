// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 0 submissions 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_0"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={Infinity}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Graded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_1"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={Infinity}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Ungraded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_2"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={Infinity}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_0"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.3333333333333333}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Graded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_1"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.3333333333333333}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Ungraded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_2"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.3333333333333333}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`render loading with null submissions 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`render loading with pending set 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`render multiple data points  1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_0"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.5}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Graded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_1"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.3333333333333333}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Ungraded
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission_dial_2"
    underlayColor="#eeeeee00"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Circle
            borderWidth={0}
            color="#00e0ff"
            formatText={[Function]}
            progress={0.16666666666666666}
            showsText={true}
            size={70}
            textStyle={
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-medium",
                "fontSize": 16,
              }
            }
            thickness={7}
            unfilledColor="#F5F5F5"
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 8,
                "textAlign": "center",
              },
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
